### PR TITLE
Appeal list filters

### PR DESCRIPTION
**Pull request by dr-bizz**
## Description

In this PR, I added functionality to the list view filter buttons. This includes exporting contacts & emails, adding contacts to appeals and deleting appeals. All models are dynamically loaded and preloaded when the mouse is hovered over the button that opens the modal in question.

#### Fixes
- Updated the panel contact number if the user searched to show how many contacts are under each appeal status.
- Added the ability to load dynamic components on the filter panel buttons on the mouse hover

#### Export contacts & emails
These models already existed, so there was less work to get them working.

#### Add Contacts Modal
Created a new modal that allows the user to add contacts to the appeal.
- Edited the`<ContactsAutocomplete/>` to exclude contact IDs so the user doesn't see contacts that are already a part of the appeal.
- Refreshes the contacts list to show new contacts in the contact list.

#### Delete Appeal Modal
Created this modal to allow the user to delete the appeal.
- after deletion, it redirects to the user back to the appeals page

## Checklist:

- [ ] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [ ] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [ ] I have requested a review from another person on the project

### CODE REVIEW

**Comment by dr-bizz** [comment]
If `codecov/patch ` is above 80% I have been approving PRs as that is a pass in my book.

File: src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx [hunk 1]
```diff
@@ -30,6 +31,16 @@ export enum TableViewModeEnum {
   Flows = 'flows',
 }
 
+export const shouldSkipContactCount = (filterPanelOpen, viewMode) => {
```
Comment (canac) [inline src/components/Tool/Appeal/AppealsContext/AppealsContext.tsx:34]: Can we add types to the arguments?

File: src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        onSubmit={onSubmit}
+      >
+        {({
+          values: { contactIds },
+          setFieldValue,
+          handleSubmit,
+          isSubmitting,
+          isValid,
+        }): ReactElement => (
+          <form onSubmit={handleSubmit} data-testid="addContactToAppealModal">
```
[the remark this answers is not in the document]
Comment (canac) [reply src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.tsx:102]: And we could change the modal title from "Add Contact(s)" to "Add Contact(s) to Appeal".

File: src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.test.tsx [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+    userEvent.click(getByRole('button', { name: 'Save' }));
+
+    await waitFor(() => {
+      expect(mockEnqueue).toHaveBeenCalledWith(
+        '2 contacts successfully added to your appeal.',
+        { variant: 'success' },
+      );
+    });
+
+    await waitFor(() => {
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.test.tsx:156]: Also fixed deleteAppealModal.text.tex

File: src/components/Tool/Appeal/Modals/DeleteAppealModal/DeleteAppealModal.tsx [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+
+export const DeleteAppealModal: React.FC<DeleteAppealModalProps> = ({
+  handleClose,
+}) => {
+  const { t } = useTranslation();
+  const { push } = useRouter();
+  const { enqueueSnackbar } = useSnackbar();
+  const [deleteAppeal] = useDeleteAppealMutation();
+  const { accountListId, appealId } = useContext(AppealsContext) as AppealsType;
+  const [isSubmitting, setIsSubmitting] = useState(false);
```
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/DeleteAppealModal/DeleteAppealModal.tsx:35]: @canac What do you mean? I didn't think you could get a `loading` variable from mutations, I thought it was only queries.

[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/DeleteAppealModal/DeleteAppealModal.tsx:35]: I didn't know you could do that. Nice

`const [deleteAppeal, { loading }] = useDeleteAppealMutation();`

File: src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (dr-bizz) [reply src/components/Tool/Appeal/Modals/AddContactToAppealModal/AddContactToAppealModal.tsx:102]: As the current one doesn't have this, I will just change the name to "Add Contact(s) to Appeal".

**Comment by dr-bizz** [comment]
The increase of files changed is due to appealInfo.graphQL being reworded. in both places.

**Review by canac — APPROVED** [review]
Great work addressing everything from the last review!

**Pull request merged**